server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala

```diff
@@ -294,6 +296,28 @@ class Serializer @Inject() (mailboxIdFactory: MailboxId.Factory) {
 
   private implicit val mailboxSetRequestReads: Reads[MailboxSetRequest] = Json.reads[MailboxSetRequest]
 
+  private implicit val vacationResponseIdReads: Reads[VacationResponseId] = Json.valueReads[VacationResponseId]
+
+  private implicit val UTCDateReads: Reads[UTCDate] = {
+    case JsString(value) =>
+      try {
+        JsSuccess(UTCDate(ZonedDateTime.parse(value, DateTimeFormatter.ISO_DATE_TIME)))
```

Comment:
you could do `Try(ZonedDateTime.parse(value, DateTimeFormatter.ISO_DATE_TIME))` and pattern match on it.
It would be more idiomatic for scala code